pkgs/applications/version-management/stgit/default.nix

```diff
@@ -1,32 +1,37 @@
 { stdenv
 , lib
+, rustPlatform
 , fetchFromGitHub
+, pkg-config
 , installShellFiles
-, python3Packages
 , asciidoc
 , docbook_xsl
 , docbook_xml_dtd_45
+, xmlto
+, libgit2
+, curl
+, bzip2
 , git
 , perl
-, xmlto
 }:
 
-python3Packages.buildPythonApplication rec {
+rustPlatform.buildRustPackage rec {
   pname = "stgit";
-  version = "1.5";
+  version = "2.0.4";
 
   src = fetchFromGitHub {
     owner = "stacked-git";
     repo = "stgit";
     rev = "v${version}";
-    sha256 = "sha256-TsJr2Riygz/DZrn6UZMPvq1tTfvl3dFEZZNq2wVj1Nw=";
+    sha256 = "sha256-ab85aX1eYuyUYR4x3Bh0pRd5xUrxrrX/9ilnBSFv/SA=";
   };
+  cargoHash = "sha256-3RuexRvsekzcACTMqZw1701eNwenKgBFHZ4Apd84Lag=";
 
-  nativeBuildInputs = [ installShellFiles asciidoc xmlto docbook_xsl docbook_xml_dtd_45 python3Packages.setuptools ];
+  nativeBuildInputs = [ pkg-config installShellFiles asciidoc xmlto docbook_xsl docbook_xml_dtd_45 perl ];
+  buildInputs = [ libgit2 curl bzip2 ];
+  propagatedBuildInputs = [ git ];
```

Comment:
Please use makeWrapper instead.

Reply:
You mean, I should keep git in `buildInputs` and run makeWrapper/wrapProgram with `--prefix PATH : ${lib.makeBinPath [ git ]}` in `installPhase` instead?

Reply:
Updated to use makeProgram